go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -219,8 +220,17 @@ func runOvnKube(ctx *cli.Context) error {
 	stopChan := make(chan struct{})
 	wg := &sync.WaitGroup{}
 
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(klog.Infof)
+	eventBroadcaster.StartRecordingToSink(
+		&corev1.EventSinkImpl{
+			Interface: ovnClientset.KubeClient.CoreV1().Events(""),
+		})
+	defer eventBroadcaster.Shutdown()
```

Comment:
When shutdown is in-progress we need to be sure there are no event writers or we will get panic. This will depend on ensuring all event writers are stopped before calling Shutdown